src/services/identity/__tests__/SitesService.spec.ts

```diff
@@ -1,78 +1,355 @@
+import mockAxios from "jest-mock-axios"
 import { ModelStatic } from "sequelize"
 
 import { Site } from "@database/models"
-
-import _SitesService from "../SitesService"
-import TokenStore from "../TokenStore"
+import {
+  repoInfo,
+  repoInfo2,
+  adminRepo,
+  noAccessRepo,
+} from "@fixtures/repoInfo"
+import {
+  mockUserWithSiteSessionData,
+  mockSessionDataEmailUser,
+  mockIsomerUserId,
+} from "@fixtures/sessionData"
+import { NotFoundError } from "@root/errors/NotFoundError"
+import type { RepositoryData } from "@root/types/repoInfo"
+import { GitHubService } from "@services/db/GitHubService"
+import { ConfigYmlService } from "@services/fileServices/YmlFileServices/ConfigYmlService"
+import IsomerAdminsService from "@services/identity/IsomerAdminsService"
+import _SitesService from "@services/identity/SitesService"
+import TokenStore from "@services/identity/TokenStore"
+import UsersService from "@services/identity/UsersService"
 
 const MockRepository = {
   findOne: jest.fn(),
 }
 
+const MockGithubService = {
+  checkHasAccess: jest.fn(),
+  getRepoInfo: jest.fn(),
+}
+
+const MockConfigYmlService = {
+  read: jest.fn(),
+}
+
+const MockUsersService = {
+  findSitesByUserId: jest.fn(),
+}
+
+const MockIsomerAdminsService = {
+  getByUserId: jest.fn(),
+}
+
 const MockTokenStore = {
   getToken: jest.fn(),
 }
 
 const SitesService = new _SitesService({
-  repository: (MockRepository as unknown) as ModelStatic<Site>,
+  siteRepository: (MockRepository as unknown) as ModelStatic<Site>,
+  gitHubService: (MockGithubService as unknown) as GitHubService,
+  configYmlService: (MockConfigYmlService as unknown) as ConfigYmlService,
+  usersService: (MockUsersService as unknown) as UsersService,
+  isomerAdminsService: (MockIsomerAdminsService as unknown) as IsomerAdminsService,
   tokenStore: (MockTokenStore as unknown) as TokenStore,
 })
 
 const mockSiteToken = "token black"
 const mockSiteName = "some site name"
+const mockSessionData = {
+  accessToken: mockSiteToken,
+  siteName: mockSiteName,
+}
 const mockSite = ({
   name: "i m a site",
   apiTokenName: "0000",
   users: [],
 } as unknown) as Site
+const genericGitHubAxiosInstance = mockAxios.create()
 
 describe("SitesService", () => {
+  beforeAll(() => {
+    // NOTE: Because SitesService uses an axios instance,
+    // we need to mock the axios instance using es5 named exports
+    // to ensure that the calls for .get() on the instance
+    // will actually return a value and not fail.
+    jest.mock("../../../services/api/AxiosInstance.ts", () => ({
+      __esModule: true, // this property makes it work
+      genericGitHubAxiosInstance,
```

Comment:
good docs! could i check if we should prefer using `__mocks__/axios.ts`?  can refer to `Sites.spec.ts` for an eg (sorry if troublesome ._.)

Reply:
Migrated in 8c2275e.